packages/pds/src/db/index.ts

```diff
@@ -179,7 +180,13 @@ export class Database {
       .transaction()
       .execute(async (txn) => {
         const dbTxn = new Database(txn, this.cfg, this.channels)
-        const txRes = await fn(dbTxn).finally(() => leakyTxPlugin.endTx())
+        const txRes = await fn(dbTxn)
+          .catch(async (err) => {
+            leakyTxPlugin.endTx()
+            await wait(10)
```

Comment:
Alright bear with me, having a thought!

Notice that in the test the query that leaks out is always the second of the two, despite the fact that both start at the same time.  The reason there is that the [`SingleConnectionProvider`](https://github.com/kysely-org/kysely/blob/a6d7bf358d5ee6552eddbd2cce6b4de5f8861a4b/src/driver/single-connection-provider.ts#L4) linearizes each query execution, essentially running a queue.  So the second query is delayed to begin until the first one completes.  You could imagine a longer chain of pending queries here, too.

I think this actually gives us a reliable way to flush the pending queries before kysely issues the rollback.  We could either perform one simple/non-effectful query like `select 1`, or (going lower-level) asking the executor to provide the connection one last time.  In either case, this completes only once other queries already on the queue have completed.  Examples:

```suggestion
            const flushQuery = sql`select 1`.execute(dbTxn.db)
            leakyTxPlugin.endTx()
            await flushQuery
```

or

```suggestion
            leakyTxPlugin.endTx()
            await dbTxn.db.getExecutor().provideConnection(async () => {})
```

Reply:
This is great @devinivy 🙌 

I ended up opting for the latter fix. Seems cleaner here, doesn't involve an extra trip, & more clear what's going on. As you said, we can upgrade it if need be

Reply:
Also buffed up the tests a bit, so there are ~20 inflight queries when the error is thrown

Reply:
Brilliant 🙏